lib/rules/no-underscore-dangle.js

```diff
@@ -162,14 +166,57 @@ module.exports = {
             }
         }
 
+        /**
+         * Check if method declaration has a underscore at the end
+         * @param {ASTNode} node node to evaluate
+         * @returns {void}
+         * @private
+         */
+        function checkForTrailingUnderscoreInMethodDefinition(node) {
+            const identifier = node.key.name;
+
+            if (typeof identifier !== "undefined" && hasTrailingUnderscore(identifier) && enforceInMethodNames) {
+                context.report({
+                    node,
+                    message: "Unexpected dangling '_' in '{{identifier}}'",
+                    data: {
+                        identifier
+                    }
+                });
+            }
+        }
+
+        /**
+         * Check if method property has a underscore at the end
+         * @param {ASTNode} node node to evaluate
+         * @returns {void}
+         * @private
+         */
+        function checkForTrailingUnderscoreInMethodProperty(node) {
+            const identifier = node.key.name;
+            const isMethod = node.method;
+
+            if (typeof identifier !== "undefined" && hasTrailingUnderscore(identifier) && enforceInMethodNames && isMethod) {
```

Comment:
Why is the check for `isMethod` necessary here? It doesn't look like this ever evaluates to falsey in the tests. Additionally, if we can remove this, I believe `checkForTrailingUnderscoreInMethodDefinition` and `checkForTrailingUnderscoreInMethodProperty` become the same function and we can just use one function for both `MethodDefinition` and `Property`.